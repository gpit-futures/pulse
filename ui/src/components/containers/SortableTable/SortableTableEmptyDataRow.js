import React from 'react';

const SortableTableEmptyDataRow = props => <tr>
  <td colSpan={props.amountCollumns}>
    {props.isLoading ? <span className="label label-default">Loading...</span> : <span className="label label-default">{props.emptyDataMessage}</span> }
  </td>
</tr>;

export default SortableTableEmptyDataRow
