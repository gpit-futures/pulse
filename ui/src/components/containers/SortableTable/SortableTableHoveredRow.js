import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import classNames from 'classnames';
import { formatNHSNumber } from '../../../utils/table-helpers/table.utils';

export default class SortableTableHoveredRow extends PureComponent {
  static propTypes = {
    rowData: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    })),
    onCellClick: PropTypes.func.isRequired,
    onMouseEnter: PropTypes.func.isRequired,
    onMouseLeave: PropTypes.func.isRequired,
    columnNameSortBy: PropTypes.string.isRequired,
    hoveredRowIndex: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  };

  render() {
    const { rowData, headers, onCellClick, columnNameSortBy, onMouseEnter, onMouseLeave, hoveredRowIndex, index, table } = this.props;
    const userId = _.flow(_.find({ name: 'id' }), _.get('value'))(rowData);
    const sourceId = _.flow(_.find({ name: 'sourceId' }), _.get('value'))(rowData);

    const rowDataItem = rowData.map((rowItem, index) => {
      if (rowItem.name === 'id') {
        return <td data-table-hover data-th={headers[index].title} key={_.uniqueId('__SortableTableHoveredRow__')} name={rowItem.name} onClick={() => onCellClick(userId, rowItem.name, sourceId)} className={classNames({ 'sorted': rowItem.name === columnNameSortBy, 'text-center': (table === 'patientsList' && rowItem.name !== 'name' && rowItem.name !== 'address') })}>{formatNHSNumber(rowItem.value)}</td>
      }
      if (rowItem.name === 'viewPatientNavigation') {
        return <td data-table-hover data-th={headers[index].title} key={_.uniqueId('__SortableTableHoveredRow__')} name={rowItem.name} onClick={() => onCellClick(userId, rowItem.name, sourceId)} className={classNames('patient-control', { 'sorted': rowItem.name === columnNameSortBy })}>{rowItem.value}</td>
      }
      return <td data-table-hover data-th={headers[index].title} key={_.uniqueId('__SortableTableHoveredRow__')} name={rowItem.name} onClick={() => onCellClick(userId, rowItem.name, sourceId)} className={classNames({ 'sorted': rowItem.name === columnNameSortBy, 'text-center': (table === 'patientsList' && rowItem.name !== 'name' && rowItem.name !== 'address') })}>{rowItem.value}</td>
    });

    return (<tr onMouseOver={() => onMouseEnter(index)} onMouseOut={() => onMouseLeave()} className={classNames({ hovered: hoveredRowIndex === index })}>
      {rowDataItem}
    </tr>)
  }
}
