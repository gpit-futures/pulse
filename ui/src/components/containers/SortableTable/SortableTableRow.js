import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import classNames from 'classnames';

export default class SortableTableRow extends PureComponent {
  static propTypes = {
    rowData: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    })),
    onCellClick: PropTypes.func.isRequired,
    columnNameSortBy: PropTypes.string.isRequired,
  };

  render() {
    const { rowData, headers, onCellClick, columnNameSortBy, id } = this.props;
    const sourceId = _.flow(_.find({ name: 'sourceId' }), _.get('value'))(rowData);

    const rowDataItem = rowData.map((rowItem, index) => {
      if (rowItem.highlighter) {
        return (
          <td data-table-hover data-th={headers[index].title} key={_.uniqueId('__SortableTableRow__')} name={rowItem.name} onClick={() => onCellClick(sourceId)} className={classNames('highlighter-wrapper', { 'sorted': rowItem.name === columnNameSortBy })}>
            <span>
              <span className={`highlighter-${rowItem.highlighter}`} />
              <span>{ rowItem.value }</span>
            </span>
          </td>
        )
      }
      return <td data-table-hover data-th={headers[index].title} key={_.uniqueId('__SortableTableRow__')} name={rowItem.name} onClick={() => onCellClick(sourceId)} className={classNames({ 'sorted': rowItem.name === columnNameSortBy })}>{rowItem.value}</td>
    });

    return (<tr className={classNames({ 'info': id === sourceId })}>
      {rowDataItem}
    </tr>)
  }
}
