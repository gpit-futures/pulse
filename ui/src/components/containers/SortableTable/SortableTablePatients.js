import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';

import SortableTableHeaderRow from './sortable-table-header-components/SortableTableHeaderRow';
import SortableTableHoveredRow from './SortableTableHoveredRow';
import SortableTableEmptyDataRow from './SortableTableEmptyDataRow';
import { getArrByTemplate } from '../../../utils/table-helpers/table.utils';

export default class SortableTablePatients extends PureComponent {
  static propTypes = {
    headers: PropTypes.arrayOf(PropTypes.object).isRequired,
    data: PropTypes.arrayOf(PropTypes.object).isRequired,
    onHeaderCellClick: PropTypes.func.isRequired,
    onCellClick: PropTypes.func.isRequired,
    sortingOrder: PropTypes.oneOf(['asc', 'desc']).isRequired,
    columnNameSortBy: PropTypes.string.isRequired,
    table: PropTypes.string.isRequired,
  };

  static defaultProps = {
    table: '',
  };

  state = {
    hoveredRowIndex: '',
  };

  getSortableTableRows = (rowsData, resourceData, emptyDataMessage) => {
    const { onCellClick, columnNameSortBy, headers, table } = this.props;
    const { hoveredRowIndex } = this.state;
    const amountCollumns = headers.length;

    return (
      _.isUndefined(resourceData)
        ? <SortableTableEmptyDataRow isLoading emptyDataMessage={emptyDataMessage} amountCollumns={amountCollumns} />
        : !resourceData.length
          ? <SortableTableEmptyDataRow isLoading={false} emptyDataMessage={emptyDataMessage} amountCollumns={amountCollumns} />
          : rowsData.map((rowData, index) =>
            <SortableTableHoveredRow
              key={_.uniqueId('__SortableTableRow__')}
              rowData={rowData}
              onCellClick={onCellClick}
              columnNameSortBy={columnNameSortBy}
              headers={headers}
              onMouseEnter={this.hoverTableRow}
              onMouseLeave={this.unHoverTableRow}
              hoveredRowIndex={hoveredRowIndex}
              index={index}
              table={table}
              resourceData={resourceData}
            />)
    )
  };

  hoverTableRow = (index) => {
    this.setState({ hoveredRowIndex: index });
  };

  unHoverTableRow = () => {
    this.setState({ hoveredRowIndex: '' });
  };

  resizeFixedTables = () => {
    const tableNames = this.tableNames;
    const tableControls = this.tableControls;
    const tableFull = this.tableFull;

    if (tableNames && tableControls && tableFull) {
      const tableFullRows = _.last(tableFull.children).children;
      const tds = _.head(tableFullRows).children;

      tableNames.style.width = `${_.head(tds).offsetWidth + 2}px`;
      tableControls.style.width = `${tds[tds.length - 1].offsetWidth + 1}px`;
    }
  };

  render() {
    const { headers, data, onHeaderCellClick, sortingOrder, columnNameSortBy, table, resourceData, emptyDataMessage } = this.props;
    const rowsData = getArrByTemplate(headers, data, '-');
    const headersName = [_.head(headers)];
    const headersView = [_.last(headers)];
    const rowsDataName = rowsData.map(el => el.filter(el => (el.name === 'name' || el.name === 'id')));
    const rowsDataView = rowsData.map(el => el.filter(el => el.name === 'viewPatientNavigation'));

    setTimeout(() => this.resizeFixedTables());
    window.addEventListener('resize', () => {
      this.resizeFixedTables()
    });

    return (
      <div>
        {(table === 'patientsList' && !_.isEmpty(data)) ? <table
          className="table table-striped  table-bordered table-sorted table-hover table-fixedcol table-patients-name"
          ref={(el) => { this.tableNames = el; }}
        >
          <colgroup>
            {/*//TODO inject theme here*/}
            {headersName.map(item => <col style={{ width: item.width }} key={_.uniqueId('__colHeadersName__')} />)}
          </colgroup>
          <thead ref="tableHead">
            <SortableTableHeaderRow
              headers={headersName}
              onHeaderCellClick={onHeaderCellClick}
              sortingOrder={sortingOrder}
              columnNameSortBy={columnNameSortBy}
            />
          </thead>
          <tbody>
            {this.getSortableTableRows(rowsDataName, resourceData, emptyDataMessage)}
          </tbody>
        </table> : null }
        <table
          className={`table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table ${table}`}
          ref={(el) => { this.tableFull = el; }}
        >
          <colgroup>
            {/*//TODO inject theme here*/}
            {headers.map((item) => {
              if (item.display) {
                return (<col style={{ width: item.width, display: item.display }} key={_.uniqueId('__colHeaders__')} />)
              }
              return (<col style={{ width: item.width }} key={_.uniqueId('__colHeaders__')} />)
            })}
          </colgroup>
          <thead>
            <SortableTableHeaderRow
              headers={headers}
              onHeaderCellClick={onHeaderCellClick}
              sortingOrder={sortingOrder}
              columnNameSortBy={columnNameSortBy}
            />
          </thead>
          <tbody>
            {this.getSortableTableRows(rowsData, resourceData, emptyDataMessage)}
          </tbody>
        </table>
        {(table === 'patientsList' && !_.isEmpty(data)) ? <table
          className="table table-striped table-bordered table-sorted table-fixedcol table-patients-controls"
          ref={(el) => { this.tableControls = el; }}
        >
          <colgroup>
            {/*//TODO inject theme here*/}
            {headersView.map(item => <col style={{ width: item.width }} key={_.uniqueId('__colHeadersView__')} />)}
          </colgroup>
          <thead>
            <SortableTableHeaderRow
              headers={headersView}
              onHeaderCellClick={onHeaderCellClick}
              sortingOrder={sortingOrder}
              columnNameSortBy={columnNameSortBy}
            />
          </thead>
          <tbody>
            {this.getSortableTableRows(rowsDataView, resourceData, emptyDataMessage)}
          </tbody>
        </table> : null }
      </div>)
  }
}

