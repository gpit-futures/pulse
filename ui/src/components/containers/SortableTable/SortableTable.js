import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';

import SortableTableHeaderRow from './sortable-table-header-components/SortableTableHeaderRow';
import SortableTableRow from './SortableTableRow';
import SortableTableEmptyDataRow from './SortableTableEmptyDataRow';
import { getArrByTemplate } from '../../../utils/table-helpers/table.utils';

export default class SortableTable extends PureComponent {
  static propTypes = {
    headers: PropTypes.arrayOf(PropTypes.object).isRequired,
    data: PropTypes.arrayOf(PropTypes.object).isRequired,
    onHeaderCellClick: PropTypes.func.isRequired,
    onCellClick: PropTypes.func.isRequired,
    sortingOrder: PropTypes.oneOf(['asc', 'desc']).isRequired,
    columnNameSortBy: PropTypes.string.isRequired,
    table: PropTypes.string.isRequired,
  };

  static defaultProps = {
    table: '',
    data: [],
  };

  getSortableTableRows = (rowsData, resourceData, emptyDataMessage) => {
    const { onCellClick, columnNameSortBy, headers, table, id } = this.props;
    const amountCollumns = headers.length - 1;
    return (
      _.isUndefined(resourceData)
        ? <SortableTableEmptyDataRow isLoading emptyDataMessage={emptyDataMessage} amountCollumns={amountCollumns} />
        : !resourceData.length
          ? <SortableTableEmptyDataRow isLoading={false} emptyDataMessage={emptyDataMessage} amountCollumns={amountCollumns} />
          : rowsData.map((rowData, index) =>
            <SortableTableRow
              key={_.uniqueId('__SortableTableRow__')}
              rowData={rowData}
              onCellClick={onCellClick}
              columnNameSortBy={columnNameSortBy}
              headers={headers}
              index={index}
              table={table}
              resourceData={resourceData}
              id={id}
            />)
    )
  };

  addHighlighters = (data, rowsData) => {
    if (!data) return;
    data.map((item, index) => {
      if (item.highlighters && item.highlighters.length) {
        item.highlighters.map((highlighter) => {
          const field = _.find({ name: highlighter.name })(rowsData[index]);
          field ? field.highlighter = highlighter.status : null;
        });
      }
    });
  };

  render() {
    const { headers, data, onHeaderCellClick, sortingOrder, columnNameSortBy, table, resourceData, emptyDataMessage } = this.props;
    const rowsData = getArrByTemplate(headers, data);

    this.addHighlighters(data, rowsData);

    return (
      <table className={`table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table ${table}`}>
        <colgroup>
          {headers.map((item) => {
            if (item.display) {
              return (<col style={{ width: item.width, display: item.display }} key={_.uniqueId('__colHeaders__')} />)
            }
            return (<col style={{ width: item.width }} key={_.uniqueId('__colHeaders__')} />)
          })}
        </colgroup>
        <thead>
          <SortableTableHeaderRow
            headers={headers}
            onHeaderCellClick={onHeaderCellClick}
            sortingOrder={sortingOrder}
            columnNameSortBy={columnNameSortBy}
          />
        </thead>
        <tbody>
          {this.getSortableTableRows(rowsData, resourceData, emptyDataMessage)}
        </tbody>
      </table>
    )
  }
}

