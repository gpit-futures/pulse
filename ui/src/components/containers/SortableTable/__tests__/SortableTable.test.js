import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import SortableTable from '../SortableTable';

Enzyme.configure({ adapter: new Adapter() })

const headers = [
  {
    key: 'cause',
    title: 'Cause',
    width: '25%',
  },
  {
    key: 'reaction',
    title: 'Reaction',
    width: '65%',
  },
  {
    display: 'none',
    key: 'sourceId',
    title: 'Source ID',
    width: '0',
  },
];
const data = [
  {
    cause: 3123123123,
    reaction: 123123,
    source: 'ethercis',
    sourceId: 'c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4',
  },
  {
    cause: 123,
    reaction: 123,
    source: 'test',
    sourceId: 'test',
  },
];
const dataWithHighlighters = [
  {
    cause: 3123123123,
    reaction: 123123,
    source: 'ethercis',
    sourceId: 'c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4',
    highlighters: [{
      name: 'cause',
      status: 'warning'
    }, {
      name: 'nonExistentField',
      status: 'warning'
    }]
  },
  {
    cause: 123,
    reaction: 123,
    source: 'test',
    sourceId: 'test',
    highlighters: [{
      name: 'cause',
      status: 'danger'
    }]
  },
];
const resourceData = [
  {
    cause: 3123123123,
    reaction: 123123,
    source: 'ethercis',
    sourceId: 'c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4',
  },
  {
    cause: 123,
    reaction: 123,
    source: 'test',
    sourceId: 'test',
  },
];

describe('Component <SortableTable />', () => {
  it('should renders with props correctly', () => {
    const sortableTable = mount(
      <SortableTable
        headers={headers}
        data={data}
        resourceData={resourceData}
        columnNameSortBy="cause"
        emptyDataMessage="No allergies"
        sortingOrder="asc"
        table=""
      />);
    expect(sortableTable).toMatchSnapshot();
  });

  it('should renders with props correctly when the resourceData is undefined', () => {
    const sortableTable = shallow(
      <SortableTable
        headers={headers}
        data={data}
        columnNameSortBy="cause"
        emptyDataMessage="No allergies"
        sortingOrder="asc"
        table=""
      />);
    expect(sortableTable).toMatchSnapshot();
  });

  it('should renders with props correctly when the resourceData is empty array', () => {
    const sortableTable = shallow(
      <SortableTable
        headers={headers}
        data={null}
        resourceData={[]}
        columnNameSortBy="cause"
        emptyDataMessage="No allergies"
        sortingOrder="asc"
        table=""
      />);
    expect(sortableTable).toMatchSnapshot();
  });

  it('should renders with props correctly with highlighters data', () => {
    const sortableTable = shallow(
      <SortableTable
        headers={headers}
        data={dataWithHighlighters}
        resourceData={resourceData}
        columnNameSortBy="cause"
        emptyDataMessage="No allergies"
        sortingOrder="asc"
        table=""
      />);
    expect(sortableTable).toMatchSnapshot();
  });
});
