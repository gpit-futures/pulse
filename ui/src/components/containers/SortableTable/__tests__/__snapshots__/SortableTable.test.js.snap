// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <SortableTable /> should renders with props correctly 1`] = `
<SortableTable
  columnNameSortBy="cause"
  data={
    Array [
      Object {
        "cause": 3123123123,
        "reaction": 123123,
        "source": "ethercis",
        "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
      },
      Object {
        "cause": 123,
        "reaction": 123,
        "source": "test",
        "sourceId": "test",
      },
    ]
  }
  emptyDataMessage="No allergies"
  headers={
    Array [
      Object {
        "key": "cause",
        "title": "Cause",
        "width": "25%",
      },
      Object {
        "key": "reaction",
        "title": "Reaction",
        "width": "65%",
      },
      Object {
        "display": "none",
        "key": "sourceId",
        "title": "Source ID",
        "width": "0",
      },
    ]
  }
  resourceData={
    Array [
      Object {
        "cause": 3123123123,
        "reaction": 123123,
        "source": "ethercis",
        "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
      },
      Object {
        "cause": 123,
        "reaction": 123,
        "source": "test",
        "sourceId": "test",
      },
    ]
  }
  sortingOrder="asc"
  table=""
>
  <table
    className="table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table "
  >
    <colgroup>
      <col
        key="__colHeaders__1"
        style={
          Object {
            "width": "25%",
          }
        }
      />
      <col
        key="__colHeaders__2"
        style={
          Object {
            "width": "65%",
          }
        }
      />
      <col
        key="__colHeaders__3"
        style={
          Object {
            "display": "none",
            "width": "0",
          }
        }
      />
    </colgroup>
    <thead>
      <SortableTableHeaderRow
        columnNameSortBy="cause"
        headers={
          Array [
            Object {
              "key": "cause",
              "title": "Cause",
              "width": "25%",
            },
            Object {
              "key": "reaction",
              "title": "Reaction",
              "width": "65%",
            },
            Object {
              "display": "none",
              "key": "sourceId",
              "title": "Source ID",
              "width": "0",
            },
          ]
        }
        sortingOrder="asc"
      >
        <tr>
          <SortableTableHeaderCell
            key="__SortableTableHeaderCell__6"
            name="cause"
            onClick={[Function]}
            sortingOrder="asc"
            title="Cause"
          >
            <th
              className="sorted asc"
              name="cause"
              onClick={[Function]}
            >
              <span>
                Cause
              </span>
               
            </th>
          </SortableTableHeaderCell>
          <SortableTableHeaderCell
            key="__SortableTableHeaderCell__7"
            name="reaction"
            onClick={[Function]}
            sortingOrder={null}
            title="Reaction"
          >
            <th
              className=""
              name="reaction"
              onClick={[Function]}
            >
              <span>
                Reaction
              </span>
               
            </th>
          </SortableTableHeaderCell>
          <SortableTableHeaderCell
            key="__SortableTableHeaderCell__8"
            name="sourceId"
            onClick={[Function]}
            sortingOrder={null}
            title="Source ID"
          >
            <th
              className=""
              name="sourceId"
              onClick={[Function]}
            >
              <span>
                Source ID
              </span>
               
            </th>
          </SortableTableHeaderCell>
        </tr>
      </SortableTableHeaderRow>
    </thead>
    <tbody>
      <SortableTableRow
        columnNameSortBy="cause"
        headers={
          Array [
            Object {
              "key": "cause",
              "title": "Cause",
              "width": "25%",
            },
            Object {
              "key": "reaction",
              "title": "Reaction",
              "width": "65%",
            },
            Object {
              "display": "none",
              "key": "sourceId",
              "title": "Source ID",
              "width": "0",
            },
          ]
        }
        index={0}
        key="__SortableTableRow__4"
        resourceData={
          Array [
            Object {
              "cause": 3123123123,
              "reaction": 123123,
              "source": "ethercis",
              "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
            },
            Object {
              "cause": 123,
              "reaction": 123,
              "source": "test",
              "sourceId": "test",
            },
          ]
        }
        rowData={
          Array [
            Object {
              "name": "cause",
              "value": 3123123123,
            },
            Object {
              "name": "reaction",
              "value": 123123,
            },
            Object {
              "name": "sourceId",
              "value": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
            },
          ]
        }
        table=""
      >
        <tr
          className=""
        >
          <td
            className="sorted"
            data-table-hover={true}
            data-th="Cause"
            key="__SortableTableRow__9"
            name="cause"
            onClick={[Function]}
          >
            3123123123
          </td>
          <td
            className=""
            data-table-hover={true}
            data-th="Reaction"
            key="__SortableTableRow__10"
            name="reaction"
            onClick={[Function]}
          >
            123123
          </td>
          <td
            className=""
            data-table-hover={true}
            data-th="Source ID"
            key="__SortableTableRow__11"
            name="sourceId"
            onClick={[Function]}
          >
            c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4
          </td>
        </tr>
      </SortableTableRow>
      <SortableTableRow
        columnNameSortBy="cause"
        headers={
          Array [
            Object {
              "key": "cause",
              "title": "Cause",
              "width": "25%",
            },
            Object {
              "key": "reaction",
              "title": "Reaction",
              "width": "65%",
            },
            Object {
              "display": "none",
              "key": "sourceId",
              "title": "Source ID",
              "width": "0",
            },
          ]
        }
        index={1}
        key="__SortableTableRow__5"
        resourceData={
          Array [
            Object {
              "cause": 3123123123,
              "reaction": 123123,
              "source": "ethercis",
              "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
            },
            Object {
              "cause": 123,
              "reaction": 123,
              "source": "test",
              "sourceId": "test",
            },
          ]
        }
        rowData={
          Array [
            Object {
              "name": "cause",
              "value": 123,
            },
            Object {
              "name": "reaction",
              "value": 123,
            },
            Object {
              "name": "sourceId",
              "value": "test",
            },
          ]
        }
        table=""
      >
        <tr
          className=""
        >
          <td
            className="sorted"
            data-table-hover={true}
            data-th="Cause"
            key="__SortableTableRow__12"
            name="cause"
            onClick={[Function]}
          >
            123
          </td>
          <td
            className=""
            data-table-hover={true}
            data-th="Reaction"
            key="__SortableTableRow__13"
            name="reaction"
            onClick={[Function]}
          >
            123
          </td>
          <td
            className=""
            data-table-hover={true}
            data-th="Source ID"
            key="__SortableTableRow__14"
            name="sourceId"
            onClick={[Function]}
          >
            test
          </td>
        </tr>
      </SortableTableRow>
    </tbody>
  </table>
</SortableTable>
`;

exports[`Component <SortableTable /> should renders with props correctly when the resourceData is empty array 1`] = `
<table
  className="table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table "
>
  <colgroup>
    <col
      key="__colHeaders__18"
      style={
        Object {
          "width": "25%",
        }
      }
    />
    <col
      key="__colHeaders__19"
      style={
        Object {
          "width": "65%",
        }
      }
    />
    <col
      key="__colHeaders__20"
      style={
        Object {
          "display": "none",
          "width": "0",
        }
      }
    />
  </colgroup>
  <thead>
    <SortableTableHeaderRow
      columnNameSortBy="cause"
      headers={
        Array [
          Object {
            "key": "cause",
            "title": "Cause",
            "width": "25%",
          },
          Object {
            "key": "reaction",
            "title": "Reaction",
            "width": "65%",
          },
          Object {
            "display": "none",
            "key": "sourceId",
            "title": "Source ID",
            "width": "0",
          },
        ]
      }
      sortingOrder="asc"
    />
  </thead>
  <tbody>
    <SortableTableEmptyDataRow
      amountCollumns={2}
      emptyDataMessage="No allergies"
      isLoading={false}
    />
  </tbody>
</table>
`;

exports[`Component <SortableTable /> should renders with props correctly when the resourceData is undefined 1`] = `
<table
  className="table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table "
>
  <colgroup>
    <col
      key="__colHeaders__15"
      style={
        Object {
          "width": "25%",
        }
      }
    />
    <col
      key="__colHeaders__16"
      style={
        Object {
          "width": "65%",
        }
      }
    />
    <col
      key="__colHeaders__17"
      style={
        Object {
          "display": "none",
          "width": "0",
        }
      }
    />
  </colgroup>
  <thead>
    <SortableTableHeaderRow
      columnNameSortBy="cause"
      headers={
        Array [
          Object {
            "key": "cause",
            "title": "Cause",
            "width": "25%",
          },
          Object {
            "key": "reaction",
            "title": "Reaction",
            "width": "65%",
          },
          Object {
            "display": "none",
            "key": "sourceId",
            "title": "Source ID",
            "width": "0",
          },
        ]
      }
      sortingOrder="asc"
    />
  </thead>
  <tbody>
    <SortableTableEmptyDataRow
      amountCollumns={2}
      emptyDataMessage="No allergies"
      isLoading={true}
    />
  </tbody>
</table>
`;

exports[`Component <SortableTable /> should renders with props correctly with highlighters data 1`] = `
<table
  className="table table-striped table-bordered table-sorted table-hover table-fixedcol table-patients-full rwd-table "
>
  <colgroup>
    <col
      key="__colHeaders__21"
      style={
        Object {
          "width": "25%",
        }
      }
    />
    <col
      key="__colHeaders__22"
      style={
        Object {
          "width": "65%",
        }
      }
    />
    <col
      key="__colHeaders__23"
      style={
        Object {
          "display": "none",
          "width": "0",
        }
      }
    />
  </colgroup>
  <thead>
    <SortableTableHeaderRow
      columnNameSortBy="cause"
      headers={
        Array [
          Object {
            "key": "cause",
            "title": "Cause",
            "width": "25%",
          },
          Object {
            "key": "reaction",
            "title": "Reaction",
            "width": "65%",
          },
          Object {
            "display": "none",
            "key": "sourceId",
            "title": "Source ID",
            "width": "0",
          },
        ]
      }
      sortingOrder="asc"
    />
  </thead>
  <tbody>
    <SortableTableRow
      columnNameSortBy="cause"
      headers={
        Array [
          Object {
            "key": "cause",
            "title": "Cause",
            "width": "25%",
          },
          Object {
            "key": "reaction",
            "title": "Reaction",
            "width": "65%",
          },
          Object {
            "display": "none",
            "key": "sourceId",
            "title": "Source ID",
            "width": "0",
          },
        ]
      }
      index={0}
      key="__SortableTableRow__24"
      resourceData={
        Array [
          Object {
            "cause": 3123123123,
            "reaction": 123123,
            "source": "ethercis",
            "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
          },
          Object {
            "cause": 123,
            "reaction": 123,
            "source": "test",
            "sourceId": "test",
          },
        ]
      }
      rowData={
        Array [
          Object {
            "highlighter": "warning",
            "name": "cause",
            "value": 3123123123,
          },
          Object {
            "name": "reaction",
            "value": 123123,
          },
          Object {
            "name": "sourceId",
            "value": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
          },
        ]
      }
      table=""
    />
    <SortableTableRow
      columnNameSortBy="cause"
      headers={
        Array [
          Object {
            "key": "cause",
            "title": "Cause",
            "width": "25%",
          },
          Object {
            "key": "reaction",
            "title": "Reaction",
            "width": "65%",
          },
          Object {
            "display": "none",
            "key": "sourceId",
            "title": "Source ID",
            "width": "0",
          },
        ]
      }
      index={1}
      key="__SortableTableRow__25"
      resourceData={
        Array [
          Object {
            "cause": 3123123123,
            "reaction": 123123,
            "source": "ethercis",
            "sourceId": "c2c286c1-2d65-4e8a-857b-1dcf3f0fa4d4",
          },
          Object {
            "cause": 123,
            "reaction": 123,
            "source": "test",
            "sourceId": "test",
          },
        ]
      }
      rowData={
        Array [
          Object {
            "highlighter": "danger",
            "name": "cause",
            "value": 123,
          },
          Object {
            "name": "reaction",
            "value": 123,
          },
          Object {
            "name": "sourceId",
            "value": "test",
          },
        ]
      }
      table=""
    />
  </tbody>
</table>
`;
